client-frameworks-support/client-support-ui5/src/ui5-support-lib.js

```diff
@@ -0,0 +1,107 @@
+sap.ui.define(['sap/ui/model/json/JSONModel', '@luigi-project/client/luigi-client'], function(JSONModel, LuigiClient) {
+  const model = new JSONModel();
+  LuigiClient.addInitListener(ctx => {
+    model.setData(ctx);
+  });
+
+  LuigiClient.addContextUpdateListener(ctx => {
+    model.setData(ctx);
+  });
+
+  return {
+    connectTo: function(oComponent) {
+      const routingConfig = oComponent.getManifestEntry('sap.ui5').routing;
+      oComponent.setModel(model, '$luigiCtx');
+
+      // Create preload view
+      sap.ui.define('luigi/DummyView', ['sap/ui/core/mvc/View', 'sap/ui/core/Control'], function(View) {
+        const DummyView = View.extend('luigi.DummyView', {
+          getControllerName: function() {},
+          createContent: function() {
+            return new sap.ui.core.HTML({ content: '<div></div>' });
+          }
+        });
+        return DummyView;
+      });
+
+      const oRouter = oComponent.getRouter();
+      oRouter.getTargets().addTarget('Preload', {
+        id: 'Preload',
+        name: 'module:luigi/DummyView',
+        type: 'View',
+        viewType: 'JS',
+        viewPath: '',
+        path: '',
+        level: 1
+      });
+      oRouter.addRoute({
+        pattern: '__preload__',
+        name: 'Preload',
```

Comment:
Maybe we anonymize this also more to avoid duplicates within the app? Same for target in the next line?